File: src/main/java/pcb/uwu/units/composite/mechanics/AccelerationUnit.java
package pcb.uwu.units.composite.mechanics;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.Unit;
import pcb.uwu.units.fundamental.TimeUnit;
import pcb.uwu.utils.ObjectCounter;

public class AccelerationUnit
		extends CompositeUnit
		implements Unit {

	public AccelerationUnit(SpeedUnit lengthUnit, TimeUnit timeUnit) {
		super(new ObjectCounter<Unit>()
				.major(lengthUnit)
				.minor(timeUnit));
	}
}

File: dimension/src/main/java/pcb/uwu/unit/derived/mechanics/AccelerationUnit.kt
package pcb.uwu.unit.derived.mechanics

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.LengthUnit
import pcb.uwu.unit.base.MassUnit
import pcb.uwu.unit.base.TimeUnit

open class AccelerationUnit : CompositeUnit
{
    constructor(lengthUnit: LengthUnit,
                timeUnit1: TimeUnit,
                timeUnit2: TimeUnit)
            : super(UnitCounter()
                        .major(lengthUnit)
                        .minor(timeUnit1)
                        .minor(timeUnit2))

    constructor(lengthUnit: SpeedUnit,
                timeUnit: TimeUnit)
            : super(UnitCounter()
                        .major(lengthUnit)
                        .minor(timeUnit))

    constructor(forceUnit: ForceUnit,
                massUnit: MassUnit)
            : super(UnitCounter()
                        .major(forceUnit)
                        .minor(massUnit))
}